mlflow/server/js/src/components/ExperimentPage.js

```diff
@@ -31,75 +34,112 @@ export class ExperimentPage extends Component {
+      .then(this.updateNextPageToken)
+      .catch(Utils.logErrorAndNotifyUser);
+  }
+
+  updateNextPageToken = (response = {}) => {
+    const { value } = response;
+    let nextPageToken = null;
+    if (value && value.next_page_token) {
+      nextPageToken = value.next_page_token;
+    }
+    this.setState({ nextPageToken, loadingMore: false });
+  };
+
+  handleLoadMoreRuns = () => {
+    const { experimentId } = this.props;
+    const { persistedState, lifecycleFilter, nextPageToken } = this.state;
+    const { orderByKey, orderByAsc, searchInput } = persistedState;
+    const orderBy = ExperimentPage.getOrderByExpr(orderByKey, orderByAsc);
+    const viewType = lifecycleFilterToRunViewType(lifecycleFilter);
+    this.setState({ loadingMore: true });
+    this.props
+      .loadMoreRunsApi(
+        [experimentId],
+        searchInput,
+        viewType,
+        orderBy,
+        nextPageToken,
+        this.loadMoreRunsRequestId,
+      )
+      .then(this.updateNextPageToken)
+      .catch(Utils.logErrorAndNotifyUser);
+  };
+
   static propTypes = {
     experimentId: PropTypes.number.isRequired,
     getExperimentApi: PropTypes.func.isRequired,
     searchRunsApi: PropTypes.func.isRequired,
+    loadMoreRunsApi: PropTypes.func.isRequired,
     history: PropTypes.object.isRequired,
     location: PropTypes.object,
   };
 
   /** Returns default values for state attributes that aren't persisted in the URL. */
   static getDefaultUnpersistedState() {
     return {
-      // String UUID associated with a GetExperiment API request
-      getExperimentRequestId: getUUID(),
-      // String UUID associated with a SearchRuns API request
-      searchRunsRequestId: getUUID(),
       // Last experiment, if any, displayed by this instance of ExperimentPage
       lastExperimentId: undefined,
       // Lifecycle filter of runs to display
       lifecycleFilter: LIFECYCLE_FILTER.ACTIVE,
     };
   }
 
-  snapshotComponentState() {
-
-  }
-
-  componentDidUpdate() {
-    this.snapshotComponentState();
+  componentDidMount() {
+    this.loadData();
   }
 
-  componentWillUnmount() {
-    // Snapshot component state on unmounts to ensure we've captured component state in cases where
-    // componentDidUpdate doesn't fire.
-    this.snapshotComponentState();
+  componentDidUpdate(prevProps) {
+    this.maybeReloadData(prevProps);
   }
 
   static getDerivedStateFromProps(props, state) {
     if (props.experimentId !== state.lastExperimentId) {
-      const newState = {
+      return {
         ...ExperimentPage.getDefaultUnpersistedState(),
         persistedState: state.lastExperimentId === undefined ?
             state.persistedState : (new ExperimentPagePersistedState()).toJSON(),
         lastExperimentId: props.experimentId,
         lifecycleFilter: LIFECYCLE_FILTER.ACTIVE,
       };
-      props.getExperimentApi(props.experimentId, newState.getExperimentRequestId);
```

Comment:
Generally, we should avoid sending requests in `getDerivedStateFromProps`. 
https://github.com/facebook/react/issues/13541#issuecomment-418101478